sale_stock_release_channel_partner_by_date/tests/test_sale_release_channel.py

```diff
@@ -22,3 +24,36 @@ def test_sale_release_channel_on_delivery_date(self):
         self.assertFalse(picking_out.release_channel_id)
         self.env["stock.release.channel"].assign_release_channel(picking_out)
         self.assertEqual(picking_out.release_channel_id, self.test_channel)
+
+    def test_inverse_release_channel_id(self):
+        test_order = self._create_sale_order(channel=self.test_channel)
+        date_ = test_order.commitment_date or test_order.expected_date
+        delivery_date = date_ and date_.date()
+
+        # A customer and shipping address has to be set first
+        with self.assertRaises(UserError):
+            test_order.partner_shipping_id = False
+            test_order.release_channel_id = self.default_channel
+
+        # Need delivery date or expected date
+        with self.assertRaises(UserError):
+            test_order.commitment_date = False
+            test_order.expected_date = False
+            test_order.release_channel_id = self.default_channel
```

Comment:
Small suggestions (not blocking here):
 - should use `self.assertRaisesRegex` instead of `self.assertRaises` in tests